homeassistant/components/zha/binary_sensor.py

```diff
@@ -33,13 +33,22 @@
 from .entity import ZhaEntity
 
 # Zigbee Cluster Library Zone Type to Home Assistant device class
-CLASS_MAPPING = {
-    0x000D: BinarySensorDeviceClass.MOTION,
-    0x0015: BinarySensorDeviceClass.OPENING,
-    0x0028: BinarySensorDeviceClass.SMOKE,
-    0x002A: BinarySensorDeviceClass.MOISTURE,
-    0x002B: BinarySensorDeviceClass.GAS,
-    0x002D: BinarySensorDeviceClass.VIBRATION,
+IAS_ZONE_CLASS_MAPPING = {
+    IasZone.ZoneType.Motion_Sensor: BinarySensorDeviceClass.MOTION,
+    IasZone.ZoneType.Contact_Switch: BinarySensorDeviceClass.OPENING,
+    IasZone.ZoneType.Fire_Sensor: BinarySensorDeviceClass.SMOKE,
+    IasZone.ZoneType.Water_Sensor: BinarySensorDeviceClass.MOISTURE,
+    IasZone.ZoneType.Carbon_Monoxide_Sensor: BinarySensorDeviceClass.GAS,
+    IasZone.ZoneType.Vibration_Movement_Sensor: BinarySensorDeviceClass.VIBRATION,
+}
+
+IAS_ZONE_NAME_MAPPING = {
+    IasZone.ZoneType.Motion_Sensor: "Motion",
+    IasZone.ZoneType.Contact_Switch: "Open",
```

Comment:
Would `Opening` be a better name? `Open` is what the state can be? would Open: Open be confusing?

Reply:
Also just noticed `Opening` is used below on line 138